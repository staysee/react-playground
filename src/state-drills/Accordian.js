import React from 'react'

class Accordian extends React.Component {
    static defaultProps = {
        sections: []
    }

    state = {
        activeSectionIndex: null
    }

    handleActiveClick(sectionIndex){
        console.log('button clicked!', {sectionIndex});
        this.setState({ activeSectionIndex: sectionIndex })
    }

    renderItem(index, section, activeSectionIndex) {
        return (
            <li key={index}>
                <button onClick={() => this.handleActiveClick(index)}>
                {section.title}
                </button>
                {(activeSectionIndex === index) && <p>{section.content}</p>}
            </li>
        )
    }

    render() {
        const { activeSectionIndex } = this.state;

        return (
            <ul className="Accordian">
                {this.props.sections.map( (section, index) => (
                    this.renderItem(index, section, activeSectionIndex)
                ))}
                {}
            </ul>
        )
    }
}

export default Accordian