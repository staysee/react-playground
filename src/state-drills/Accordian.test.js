import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordian from './Accordian'

describe('Accordian Component', () => {
    const sectionsProp = [
        {
          title: 'Section 1',
          content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
        },
        {
          title: 'Section 2',
          content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
        },
        {
          title: 'Section 3',
          content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
        },
    ]

    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders an empty li given no section', () => {
		const wrapper = shallow(<Accordian />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

    it('renders no sections as active by default', () => {
		const wrapper = shallow(<Accordian sections={sectionsProp} />)
		expect(toJson(wrapper)).toMatchSnapshot()
    })
    
    it('opens a clicked section', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('opens only the last clicked section when multiple sections have been clicked', () => {
        const wrapper = shallow(<Accordian sections={sectionsProp} />)
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})