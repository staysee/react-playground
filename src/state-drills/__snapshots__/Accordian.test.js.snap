// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component opens a clicked section 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordian Component opens only the last clicked section when multiple sections have been clicked 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`Accordian Component renders an empty li given no section 1`] = `
<ul
  className="Accordian"
/>
`;

exports[`Accordian Component renders no sections as active by default 1`] = `
<ul
  className="Accordian"
>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
